src/backend/marsha/core/tests/test_api_video.py

```diff
@@ -1491,6 +1491,40 @@ def test_api_video_update_live_state(self):
             {"cloudwatch": {"logGroupName": "/aws/lambda/dev-test-marsha-medialive"}},
         )
 
+    @override_settings(UPDATE_STATE_SHARED_SECRETS=["shared secret"])
+    def test_api_video_update_live_state_stopped(self):
+        """Updating state to stopped should delete all the AWS elemental stack."""
+        video = VideoFactory(
+            id="a1a21411-bf2f-4926-b97f-3c48a124d528",
+            upload_state=PENDING,
+            live_state=RUNNING,
+            live_info={},
+        )
+        data = {
+            "logGroupName": "/aws/lambda/dev-test-marsha-medialive",
+            "state": "stopped",
+        }
+
+        with mock.patch('marsha.core.api.delete_aws_element_stack') as delete_aws_element_stack_mock:
+            response = self.client.patch(
+                "/api/videos/{!s}/update-live-state/".format(video.id),
+                data,
+                content_type="application/json",
+                HTTP_X_MARSHA_SIGNATURE=(
+                    "6b551203cd79200fe9f02e20e69bb75a583dfe833c997a849b8cf59802185199"
+                ),
+            )
+            delete_aws_element_stack_mock.assert_called_once()
+        video.refresh_from_db()
+
+        self.assertEqual(response.status_code, 200)
```

Comment:
so if something fails while calling AWS to delete for example, the video state will not be changed (good) but what will the API call get as error ? and how will it handle it ? retry ? The error will be logged in sentry but I'm not sure what will be the consequences of this failure on the pipeline.

Reply:
The API call will receive a 500 and the lambda will retry again 5 times.